src/components/SplitCsvUpload/SplitCsvUpload.tsx

```diff
@@ -0,0 +1,104 @@
+import { UploadOutlined } from '@ant-design/icons'
+import { Trans } from '@lingui/macro'
+import TooltipIcon from 'components/TooltipIcon'
+import { ThemeContext } from 'contexts/themeContext'
+import { Split } from 'models/v2/splits'
+import { ChangeEventHandler, useContext } from 'react'
+import { splitPercentFrom } from 'utils/v2/math'
+
+// https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readyState
+const READY_STATE_DONE = 2
+
+const readCsvFile = async (file: File): Promise<string | null> => {
+  const reader = new FileReader()
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function (evt) {
+      if (evt?.target?.readyState != READY_STATE_DONE) return resolve(null)
+      if (evt?.target.error) {
+        reject(evt.target.error)
+        return
+      }
+
+      const content = evt.target.result as string | null
+
+      resolve(content)
+    }
+
+    reader.readAsText(file)
+  })
+}
+
+const parseCsvFile = (csvContent: string) => {
+  const [, ...rows] = csvContent.split('\n')
+
+  const splits: Split[] = rows.map(row => {
+    const [
```

Comment:
maybe it's worth kicking the reader out to something more general and having typed models for the kinds of CSV's we're expecting, i assume there are other places in the UI where CSV upload is on the roadmap

Reply:
Totally. I think lets keep the parser in this file for now, but I've moved the file reader to a `utils/file.ts` folder. 🙏

Reply:
yup exactly

parser where the parsing happens, but reader goes somewhere that it can be reused 💯